### PR TITLE
Lesson 07 

### PR DESCRIPTION
**Pull request by elnurad**

### CODE REVIEW

File: students/elnurad/lesson07/create_personjob.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    """
+        This class defines Department and maintains department number,
+        department name, name of the manager and duration of a job.
+    """
+
+    logger.info('The Department class to define department')
+    logger.info("Department number")
+    dept_number = CharField(max_length = 4)
+    logger.info("Department Name")
+    dept_name = CharField(primary_key = True, max_length = 30)
```
Comment (ldconejo) [inline students/elnurad/lesson07/create_personjob.py]: The primary key for this one should be either _dept_number_ or a key generated automatically.

File: students/elnurad/lesson07/create_personjob.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        held by a Person.
+    """
+
+    job_name = CharField(primary_key = True, max_length = 30)
+    start_date = DateField(formats = 'YYYY-MM-DD') 
+    end_date = DateField(formats = 'YYYY-MM-DD')   
+    salary = DecimalField(max_digits = 7, decimal_places = 2)  
+    person_employed = ForeignKeyField(Person, related_name='was_filled_by', null = False)
+    logger.info('Job Department')
+    job_department = ForeignKeyField(Department, related_name = 'departments', null = False)
```
Comment (ldconejo) [inline students/elnurad/lesson07/create_personjob.py]: The Job table should have the deparment number as foreign key from the Department table. The idea, for assignment purposes, is that you learn to combine data from the two tables, Job and Department, to print it on screen.

**Comment by ldconejo** [comment]
Hi Elnura,

There are a couple of changes that you need to make to your database structure and how the printed data is generated. The general idea is that, when you're printing job data, you use the department code to fetch de department name from the Department table. It's pretty common, when working with SQL databases, to have to combine data from multiple tables.

Once those changes are ready, go ahead and submit a new pull request.

Regards,

Luis

File: students/elnurad/lesson07/mailroomdb/mailroom_main.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        for donor in query:
+            print(donor.donor_name)
+    except Exception as e:
+        logger.info(e)
+    finally:
+        logger.info('database closed')
+        database.close()
+
+
+def update_donor():
```
Comment (ldconejo) [inline students/elnurad/lesson07/mailroomdb/mailroom_main.py]: For this one, the update should either update the name of the donor or the value of an existing donation. The way it is implemented right now, it simply adds a new donation.

**Comment by ldconejo** [comment]
For the mailroom part, adding and deleting are working ok. The update function needs to be updated to modify existing data in the database, either the name of the donor or the value of an existing donation (or both).

-Luis

**Comment by elnurad** [comment]
Hi Luis,

Thank you for all the feedback. I just created a new pull request with
updated files. Please let me know if they look okay.

Sincerely,
Elnura

**Pull request merged**